### PR TITLE
[SYCL] Improve diagnostics for invalid work group size with L0 backend.

### PR DESCRIPTION
**Pull request by maarquitos14**
Prior to this change, when requesting more work items than available while using L0 as backend, the error message was quite vague and didn't really help users understand the problem.

### CODE REVIEW

**Comment by maarquitos14** [comment]
Friendly ping @sergey-semenov @intel/llvm-reviewers-runtime

**Comment by sergey-semenov** [comment]
LGTM. Could you add a test that checks that this is the thrown exception for the expected backends?

**Comment by maarquitos14** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1611

**Comment by maarquitos14** [comment]
Since I opened this PR, my local server has updated from OpenCL2.0 to OpenCL3.0. For this reason, I noticed that this part of the code was not handling versions newer than 2.0 for OpenCL. I believe there are no changes wrt what we use in this piece of code between v2.0 and v3.0, so I simply extended the behavior of v2.0 to 3.0. However, I think this requires further review @sergey-semenov.

**Comment by maarquitos14** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1611

**Comment by maarquitos14** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1611

File: sycl/source/detail/error_handling/enqueue_kernel.cpp [hunk 1]
```diff
@@ -51,12 +51,17 @@ void handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
   // versions.  If this is an OpenCL backend, get the version.
   bool IsOpenCL = false;    // Backend is any OpenCL version
   bool IsOpenCLV1x = false; // Backend is OpenCL 1.x
-  bool IsOpenCLV20 = false; // Backend is OpenCL 2.0
-  if (Platform.get_backend() == sycl::backend::opencl) {
+  bool IsOpenCLVGE20 = false; // Backend is Greater or Equal to OpenCL 2.0
+  bool IsL0 = false;          // Backend is any OneAPI Level 0 version
```
Comment (dm-vodopyanov) [inline sycl/source/detail/error_handling/enqueue_kernel.cpp:55]: nit:
```suggestion
  bool IsLevelZero = false;          // Backend is any OneAPI Level Zero version
```

**Review by sergey-semenov — APPROVED** [review]

**Comment by maarquitos14** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1611

**Comment by steffenlarsen** [comment]
CUDA Failed Tests (1):
    SYCL :: SubGroup/shuffle_fp64.cpp
Reported in https://github.com/intel/llvm/issues/8516

**Pull request merged**